src/simtools/model/legacy_model_parameter.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/python3
+"""
+Handling of legacy model parameters for backward compatibility.
+
+Collects routines to convert legacy model parameters to more recent
+formats. This is a fine-tuned process and requires to hard wire the
+changes. All code related to legacy model parameters should go into
+this module.
+
+"""
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+UPDATE_HANDLERS = {}
+
+
+def register_update(name):
+    """Register update handler for legacy model parameter."""
+
+    def deco(func):
+        UPDATE_HANDLERS[name] = func
+        return func
+
+    return deco
+
+
+def apply_legacy_updates_to_parameters(parameters, legacy_updates):
```

Comment:
Wondering here what would happen if new keys are introduced or removed (such as unit, type, value) in new schemas.